src/app/fetch-stories.tsx

```diff
@@ -0,0 +1,37 @@
+"use server";
+
+export interface fetchInterface {
+    page: number,
+    fetchTopStories: boolean
+}
+
+export async function fetchStories(fetchData: fetchInterface){
+    const perPage = 48
+    let url = ""
+
+    if (fetchData.fetchTopStories){
+        url = "https://hacker-news.firebaseio.com/v0/topstories.json"
+    }
+    else {
+        url = "https://hacker-news.firebaseio.com/v0/newstories.json"
+    }
+
+    const stories:number[] = await fetch(url, {
+        next: {
+            revalidate: 120
+        }
+    }).then(res => res.json());
+
+    const start = fetchData.page*perPage;
+    const end = start + perPage;
+    const storiesToFetch = stories.slice(start, end);
+    const hasMoreStories = storiesToFetch.length > 0;
```

Comment:
```suggestion
    const hasMoreStories = storiesToFetch.length == perPage;
```

Changing it to this would detect the last page earlier, if ex the last page is only returning 5 items.

Reply:
Fixed! :D